api_app/analyzers_manager/file_analyzers/cape_sandbox.py

```diff
@@ -0,0 +1,246 @@
+                response_json in this case should look somewhat like this:
+
+            {
+                'error': True,
+                'error_value': 'Error adding task to database',
+                'errors': [{
+                    'filename.exe':
+                        'Not unique, as unique option set on submit or in conf/web.conf'
+                }]
+            }
+            """
+
+            logger.info(
+                f"Job: {self.job_id} -> "
+                "File uploaded is already present in the database. "
+                "Querying its information through it's md5 hash.."
+            )
+
+            status_id = self.__search_by_md5()
+            gui_report_url = self.__base_url + "/submit/status/" + status_id
+            report_url = (
+                self.__base_url + "/apiv2/tasks/get/report/" + status_id + "/json"
+            )
+            to_respond["result_url"] = gui_report_url
+
+            try:
+                final_request = self.__session.get(
+                    report_url,
+                )
+            except requests.RequestException as e:
+                raise AnalyzerRunException(e)
+
+            to_respond["response"] = final_request.json()
+
+        return to_respond
+
+    def __search_by_md5(self) -> str:
+        db_search_url = self.__base_url + "/apiv2/tasks/search/md5/" + self.md5
+
+        try:
+            q = self.__session.get(db_search_url)
+            q.raise_for_status()
+
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        data_list = q.json().get("data")
+        if not data_list:
+            raise AnalyzerRunException(
+                "'data' key in response isn't populated in __search_by_md5 as expected"
+            )
+
+        status_id_int = data_list[0].get("id")
+        status_id = str(status_id_int)
+        return status_id
+
+    def __poll_for_result(
+        self,
+        task_id,
+    ) -> dict:
+        error_in_poll = False
+        status_api = self.__base_url + "/apiv2/tasks/status/" + str(task_id)
+        for i in range(self.max_tries):
+            error_details = None
+
+            logger.info(
+                f" Job: {self.job_id} -> "
+                f"Starting poll number #{i + 1}/{self.max_tries}"
+            )
+            try:
+                r = self.__session.get(status_api)
+                if r.status_code == 429:
+                    logger.info(
+                        f"Job: {self.job_id} -> "
+                        "Rate limited by CAPESandbox API. "
+                        f"Sleeping for {self.poll_distance} seconds."
+                    )
+
+                else:
+                    r.raise_for_status()
+
+                error_in_poll = False
+
+            except requests.RequestException as e:
+                error_in_poll = True
+                error_details = e
+
+                logger.exception(
```

Comment:
this cannot be at `exception` level because this is not something not problematic. It is normal that this would happen

Reply:
Okay I will simply log it then?